atlasdb-cassandra/src/main/java/com/palantir/atlasdb/cassandra/ReloadingCqlClusterContainer.java

```diff
@@ -54,23 +54,19 @@ public final class ReloadingCqlClusterContainer implements Closeable, Supplier<C
     private boolean isClosed;
 
     private ReloadingCqlClusterContainer(
-            CassandraClusterConfig cassandraClusterConfig,
             Refreshable<CassandraServersConfig> refreshableCassandraServersConfig,
-            Namespace namespace,
             CqlClusterFactory cqlClusterFactory) {
         this.isClosed = false;
         this.currentCqlCluster = new AtomicReference<>(Optional.empty());
-        this.refreshableCqlCluster = refreshableCassandraServersConfig.map(cassandraServersConfig ->
-                createNewCluster(cassandraClusterConfig, cassandraServersConfig, namespace, cqlClusterFactory));
+        this.refreshableCqlCluster = refreshableCassandraServersConfig.map(
+                cassandraServersConfig -> createNewCluster(cassandraServersConfig, cqlClusterFactory));
 
         this.refreshableSubscriptionDisposable = refreshableCqlCluster.subscribe(cqlCluster -> {
-            Optional<CqlCluster> maybeCqlClusterToClose = currentCqlCluster.getAndSet(Optional.of(cqlCluster));
-            if (maybeCqlClusterToClose.isPresent()) {
-                try {
-                    maybeCqlClusterToClose.get().close();
-                } catch (IOException e) {
-                    log.warn("Failed to close CQL Cluster after reloading server list", e);
-                }
+            Optional<CqlCluster> maybeClusterToClose = currentCqlCluster.getAndSet(Optional.of(cqlCluster));
+            try {
+                shutdownCluster(maybeClusterToClose);
+            } catch (IOException e) {
+                log.warn("Failed to close CQL Cluster. This may result in a resource leak", e);
```

Comment:
I'll be honest; I thought you'd factor out the try catch block and use that in both places